app/web/features/messages/groupchats/MuteDialog.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+  DialogProps,
+  FormControl,
+  FormControlLabel,
+  Radio,
+  RadioGroup,
+} from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+} from "components/Dialog";
+import { groupChatKey, groupChatsListKey } from "features/queryKeys";
+import { RpcError } from "grpc-web";
+import { useTranslation } from "i18n";
+import { GLOBAL, MESSAGES } from "i18n/namespaces";
+import React, { useState } from "react";
+import { useMutation, useQueryClient } from "react-query";
+import { service } from "service";
+import dayjs from "utils/dayjs";
+
+type DurationChoice = "1h" | "8h" | "1d" | "1w" | "1m" | "forever";
+
+export default function MuteDialog({
+  groupChatId,
+  ...props
+}: DialogProps & { groupChatId: number }) {
+  const { t } = useTranslation([GLOBAL, MESSAGES]);
+  const queryClient = useQueryClient();
+  const muteMutation = useMutation<void, RpcError, DurationChoice>(
+    async (duration) => {
+      let d = null;
+      if (duration === "1h") d = dayjs.duration({ hours: 1 });
+      else if (duration === "8h") d = dayjs.duration({ hours: 8 });
+      else if (duration === "1d") d = dayjs.duration({ days: 1 });
+      else if (duration === "1w") d = dayjs.duration({ weeks: 1 });
+      else if (duration === "1m") d = dayjs.duration({ months: 1 });
+      await service.conversations.muteChat({
+        groupChatId,
+        forDuration: d ?? undefined,
+        forever: !d,
+      });
```

Comment:
```suggestion
      let d;
      if (duration === "1h") d = dayjs.duration({ hours: 1 });
      else if (duration === "8h") d = dayjs.duration({ hours: 8 });
      else if (duration === "1d") d = dayjs.duration({ days: 1 });
      else if (duration === "1w") d = dayjs.duration({ weeks: 1 });
      else if (duration === "1m") d = dayjs.duration({ months: 1 });
      await service.conversations.muteChat({
        groupChatId,
        forDuration: d,
        forever: !d,
      });
```
Pretty minor thing, but I find deliberately setting something to `undefined` a bit weird since it's meant to be completely missing, but not a big deal here I suppose